linux/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GnomeThemeMonitorService.java

```diff
@@ -0,0 +1,82 @@
+package com.github.weisj.darkmode.platform.linux.gnome;
+
+import com.github.weisj.darkmode.platform.ThemeMonitorService;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.github.weisj.darkmode.platform.linux.gnome.GtkVariants.guessFrom;
+import static java.util.Objects.isNull;
+
+public class GnomeThemeMonitorService implements ThemeMonitorService {
+    private final Logger logger = Logger.getLogger(GnomeThemeMonitorService.class.getName());
+    private boolean initializedGnomeGtkThemeMonitorSuccessfully;
+    private GnomeGtkThemeMonitor gnomeGtkThemeMonitor;
+    private Thread gnomeGtkThemeMonitorThread;
+
+    public GnomeThemeMonitorService() {
+        try {
+            gnomeGtkThemeMonitor = new GnomeGtkThemeMonitor();
+            initializedGnomeGtkThemeMonitorSuccessfully = true;
+        } catch (IOException | InterruptedException e) {
+            initializedGnomeGtkThemeMonitorSuccessfully = false;
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean isDarkThemeEnabled() {
+        //TODO: Stop guessing and check the settings when available (like what is mentioned in the GtkVariants class)
+        return gnomeGtkThemeMonitor.readGtkTheme().equals(guessFrom(gnomeGtkThemeMonitor.readGtkTheme()).get("night"));
+    }
+
+    @Override
+    public boolean isHighContrastEnabled() {
```

Comment:
This can stay as it is for now. Maybe `ThemeMonitorService` could return a sort of `Capability` information that is used to determine which part of the settings panel to enable/disable. This way it's clear to the user that high contrast mode isn't supported at the moment.

Reply:
I agree. This should come with the next PR that adds the area to the LAF settings for setting which of the user's themes is their light and dark themes.